### ISSUE TITLE
`winget upgrade pnpm` will remove the PATH FROM user env path, and won't added the new PATH to user env path again.

### ISSUE THREAD

**Issue by gotounix** [pnpm/pnpm#8077]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

any version

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Reproduction steps

```
winget upgrade pnpm.pnpm
```

### Describe the Bug

While using `winget upgrade pnpm` to upgrade pnpm, the newest version will install successful, but it will remove the PATH FROM user env path, and won't added the new PATH to user env path again.
So the `pnpm` command will not find.

### Expected Behavior

Added the new PATH to user env path.

### Which Node.js version are you using?

any

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by josh-hemphill** [comment]
I had come to report this same issue. Winget works great to keep my pnpm exe updated since winget comes pre-installed on windows now (I keep the rest of the installation elsewhere and just delete the old exe from there so it doesn't take precedence in the PATH), except for this issue; I have to uninstall and reinstall to get the path added back.

**Comment by gotounix** [comment]
@josh-hemphill It may be a bug of `winget`.
https://github.com/microsoft/winget-cli/issues/4044

**Comment by josh-hemphill** [comment]
Yep, just forgot to link that issue. There's also https://github.com/microsoft/winget-cli/issues/3616

**Comment by zkochan** [comment]
Closing: this is not something pnpm can fix.

What the WinGet package writes to the user's PATH is decided by that package's manifest, not by pnpm.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
